reset.c

```diff
@@ -0,0 +1,888 @@
+    khash_t(aux_set) *auxtagsin;    //SET that holds the aux tags to be reatined
+    khash_t(pg_set) *pgout;         //SET that holds the pg tags to be removed
+    int tmpfield;                   //tmp field
+} conf_data;
+
+/// usage - print the usage
+/** @param fp pointer to the file / terminal to which usage to be dumped
+returns nothing
+*/
+static void usage(FILE *fp)
+{
+    fprintf(fp,
+            "Usage: samtools reset [options]\n\
+      -i <file path>        input file\n\
+      -o <file path>        output file\n\
+      -k <file path>        aux tags to be retained in csv file, optional\n\
+      -r <file path>        aux tags to be rejected in csv file, optional\n\
+      --reject-PG [<file path>] pg names to be rejected in csv file, optional\n\
+      --no-RG               to have RG lines or not, optional\n");
+
+    sam_global_opt_help(fp, "--.--.--");
+    return;
+}
+
+/// getdefaultauxtags - add default aux tags to the collection
+/** @param config pointer to conf_data which holds the aux_set pointer; if the aux_set exists, clear and create new one
+returns 1 on failure and 0 on success
+*/
+int getdefaultauxtags(conf_data *config)
+{
+    int ret = 0;
+
+    if (!config) {
+        return 1;
+    }
+
+    if (config->auxtagsin) {      //free if already exists
+        kh_destroy(aux_set, config->auxtagsin);
+        config->auxtagsin = NULL;
+    }
+
+    config->auxtagsin = kh_init(aux_set);    //initialize the new one
+
+    //add defaults, they may get overridden by reject option
+    kh_put(aux_set, config->auxtagsin, TAGNUM("BC"), &ret);   //barcode
+    kh_put(aux_set, config->auxtagsin, TAGNUM("CB"), &ret);   //cell identifier
+    //TODO add tags which are kept for backward compatibility?
+    kh_put(aux_set, config->auxtagsin, TAGNUM("LB"), &ret);   //library
+    kh_put(aux_set, config->auxtagsin, TAGNUM("MI"), &ret);   //molecular identifier
+
+    return 0;
+}
+
+/// getcsvauxtags - aux tag parsed from csv input file, validate and add to internal set
+/** @param field - the aux tag parsed
+ *  @param data - pointer to conf_data which holds internal flags and aux_set
+returns -1 on failure and 0 on success
+*/
+int getcsvauxtags(const char *field, void *data)
+{
+    int ret = 0, len = 0;
+    conf_data *config = (conf_data*)data;
+
+    if (!field || !config || !config->auxtagsin)
+        return -1;
+
+    len = strlen(field);
+    if (len != 2)                                                       //invalid tag!
+        return -1;
+
+    if (!config->tmpfield)                                              //tmpfield says add or remove
+        kh_put(aux_set, config->auxtagsin, TAGNUM(field), &ret);        //add to set
+    else {                                                              //remove from set
+            khint_t iter = kh_get(aux_set, config->auxtagsin, TAGNUM(field));
+            iter == kh_end(config->auxtagsin) ? 0 : kh_del(aux_set, config->auxtagsin, iter);
+    }
```

Comment:
Inconsistent indentation.  Please keep to 4 spaces per level.

Also (personal preference), I dislike mixed bracing and non-bracing.  Ie I'd replace

```
if (expr)
    blah1;
else {
    blah2;
    blah3
}
```

with

```
if (expr) {
    blah1;
} else {
    blah2;
    blah3
}
```

We don't really have a style guide though, so I don't know how others think?  It's certainly not wrong, and we probably have lots of the former code style already present - so this is more for discussion than a request to change.

Reply:
Done.